crates/typed-store/src/rocks/mod.rs

```diff
@@ -94,11 +112,40 @@ pub struct DBMap<K, V> {
     db_metrics: Arc<DBMetrics>,
     read_sample_interval: SamplingInterval,
     write_sample_interval: SamplingInterval,
+    _cf_metric_reporter: Arc<JoinHandle<()>>,
+    exit_signal: Arc<AtomicBool>,
 }
 
 unsafe impl<K: Send, V: Send> Send for DBMap<K, V> {}
 
 impl<K, V> DBMap<K, V> {
+    pub(crate) fn new(
+        db: Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
+        opt_cf: &str,
+        db_metrics: Arc<DBMetrics>,
+    ) -> Self {
+        let exit = Arc::new(AtomicBool::new(false));
+        let exit_cloned = exit.clone();
+        let db_cloned = db.clone();
+        let db_metrics_cloned = db_metrics.clone();
+        let cf = opt_cf.to_string();
+        let _cf_metric_reporter = Arc::new(std::thread::spawn(move || {
```

Comment:
I think it would probably be better to use a tokio task for this, if we're assuming that a DB is already always going to be constructed in a a tokio runtime. The ergonomics and capabilities are going to dramatically improve how this metrics reporter task operate.